contracts/Certificate.sol

```diff
@@ -73,7 +73,7 @@ contract Certificate is
     // todo require _sources[_latestTokenId][n] doesnt exist
     for (uint256 i = 0; i < removalIds.length; i++) {
       if (removalAmounts[i] == 0) {
-        break;
+        revert("Certificate: Removal amount 0");
```

Comment:
we should prefer [custom errors](https://blog.soliditylang.org/2021/04/21/custom-errors/) moving forward